### PR TITLE
Problem: ZMQ doesn't expose the MULTICAST_LOOP socket option

### PR DESCRIPTION
**Pull request by simias**
I attempted to add multicast tests as well. I added a method `is_multicast_available` that attempts to send and receive a multicast datagram, if it fails the ZMQ multicast tests are not run.

### CODE REVIEW

**Comment by bluca** [comment]
Could you please document the option in doc/zmq_[g|s]etsockopt ? Also you need to add the define in src/zmq_drafts.h

**Comment by simias** [comment]
I've also updated `zmq_udp(7)` with the new additions

**Comment by bluca** [comment]
doc/zmq_getsockopt.txt too please

File: doc/zmq_getsockopt.txt [hunk 1]
```diff
@@ -892,6 +892,18 @@ Default value:: -1
...
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Retrieve the current multicast loopback configuration. A value of `1`
+means that the multicast packets sent on this socket will be looped
+back to local listening interface.
+
+[horizontal]
+Option value type:: int
+Option value unit:: 0, 1
+Default value:: 1
+Applicable socket types:: ZMQ_RADIO
```
Comment (bluca) [inline doc/zmq_getsockopt.txt]: perhaps ```ZMQ_RADIO, when using UDP transport``` ? Otherwise it might get confusing with pgm/norm

Comment (simias) [reply doc/zmq_getsockopt.txt]: Yeah I wasn't sure about that, there's also the matter of raw sockets which I haven't quite understood yet.

Comment (bluca) [reply doc/zmq_getsockopt.txt]: The raw socket, for UDP, it's about the ZMQ_DGRAM type - it's really undocumented, but IIRC it's the equivalent of ZMQ_STREAM and it just translates zmq messages into raw datagrams (or TCP stream) without any ZMTP header.

**Comment by simias** [comment]
Looks like the buildbot didn't run with my latest changes.

**Comment by simias** [comment]
Oh, nevermind. It didn't refresh for some reason.

**Comment by bluca** [comment]
@simias there are compilation errors on CentOS 6 with gcc 4.4:

```
[  204s] tests/test_radio_dish.cpp: In function 'bool is_multicast_available(int)':
[  204s] tests/test_radio_dish.cpp:301: error: dereferencing pointer 'any_ipv6' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:294: error: dereferencing pointer 'any_ipv6' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:293: error: dereferencing pointer 'any_ipv6' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:291: error: dereferencing pointer 'any_ipv6' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:292: error: dereferencing pointer 'any_ipv6' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:288: note: initialized from here
[  204s] tests/test_radio_dish.cpp:301: error: dereferencing pointer 'mcast_ipv6' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:289: note: initialized from here
[  204s] tests/test_radio_dish.cpp:321: error: dereferencing pointer 'any_ipv4' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:313: error: dereferencing pointer 'any_ipv4' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:314: error: dereferencing pointer 'any_ipv4' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:310: note: initialized from here
[  204s] tests/test_radio_dish.cpp:321: error: dereferencing pointer 'mcast_ipv4' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:311: note: initialized from here
[  204s] tests/test_radio_dish.cpp:350: error: dereferencing pointer 'mcast_ipv6' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:348: note: initialized from here
[  204s] tests/test_radio_dish.cpp:369: error: dereferencing pointer 'mcast_ipv4' does break strict-aliasing rules
[  204s] tests/test_radio_dish.cpp:367: note: initialized from here
[  204s] make[2]: *** [tests/tests_test_radio_dish-test_radio_dish.o] Error 1
```
https://build.opensuse.org/package/live_build_log/network:messaging:zeromq:git-draft/libzmq/CentOS_6/x86_64

**Comment by simias** [comment]
I mean they're not wrong but given the sockaddr "inheritance" scheme it's hard to respect strict aliasing. I'm going to see if I can work around that.

**Comment by simias** [comment]
GCC 7.3.0 with `-fstrict-aliasing -Wstrict-aliasing=1 -pedantic -Wall -Wextra` doesn't see anything wrong with the code, that's convenient.

**Pull request merged**